domenetjenester/iay/src/main/java/no/nav/foreldrepenger/abakus/registerdata/ytelse/infotrygd/beregningsgrunnlag/YtelseBeregningsgrunnlagArbeidsforhold.java

```diff
@@ -1,27 +1,47 @@
 package no.nav.foreldrepenger.abakus.registerdata.ytelse.infotrygd.beregningsgrunnlag;
 
 import java.math.BigDecimal;
+import java.util.Map;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 import no.nav.foreldrepenger.abakus.domene.iay.kodeverk.InntektPeriodeType;
-import no.nav.foreldrepenger.abakus.kodeverk.KodeverkRepository;
 import no.nav.foreldrepenger.abakus.typer.OrganisasjonsNummerValidator;
 import no.nav.tjeneste.virksomhet.infotrygdberegningsgrunnlag.v1.informasjon.Arbeidsforhold;
 
 public class YtelseBeregningsgrunnlagArbeidsforhold {
+
+    private static final String DAGLIG_KILDE = "D";
+    private static final String UKENTLIG_KILDE = "U";
+    private static final String BIUKENTLIG_KILDE = "F";
+    private static final String MÅNEDLIG_KILDE = "M";
+    private static final String ÅRLIG_KILDE = "Å";
+    private static final String FASTSATT25PAVVIK_KILDE = "X";
+    private static final String PREMIEGRUNNLAG_KILDE = "Y";
+
+    private static final Map<String, InntektPeriodeType> INNTEKT_PERIODE_TYPE_MAP = Map.of(
```

Comment:
Ta heller kilde kode so minput til InntektPeriodeType og generer map der?  Slipper da å registrere alle entries, og kan garantere at ingen er uteglemt eller overlappende

Reply:
Vi har ett behov som er mapping IT->grunnlag, men siden dette kodeverket koden brukes andre steder, så hører dette hjemme i mappingen. 
Tror vi har blandet litt mye tidligere slik at TRex/Arena-koder er behandlet som "offisielle"

Reply:
InntektPeriodeType er igjen i bruk andre steder enn for ytelse. Kodene er kanskje initielt motivert av Infotrygd, men heller mot at kodene vi får fra kilden står her og integrasjonen vedlikeholdes i mappingen til ACL-objekter